src/service/resources-impl.js

```diff
@@ -435,6 +435,69 @@ export class Resources {
   }
 
   /**
+   * Runs the specified subtree "show" mutation and ensures that measures
+   * and layouts performed for the affected elements.
+   * @param {!Element} root
+   * @param {function()} mutator
+   * @return {!Promise}
+   */
+  showSubtree(root, mutator) {
```

Comment:
My feeling is that we're better off with an API like:

`#notifyPositionChange`

and ResourceManager just assumes it has to remeasure everything in the doc. That should be in the end pretty cheap after the first measurement.

The contract would be that the caller calls it when the DOM is in steady state from its POV.